crypto/evp/digest.c

```diff
@@ -225,7 +223,7 @@ int EVP_DigestInit_ex(EVP_MD_CTX *ctx, const EVP_MD *type, ENGINE *impl)
 #endif
     if (ctx->digest != type) {
         if (ctx->digest && ctx->digest->ctx_size)
-            OPENSSL_free(ctx->md_data);
+            OPENSSL_clear_free(ctx->md_data, ctx->digest->ctx_size);
```

Comment:
I'm going to assume that if ctx_size != 0 that md_data != NULL, but maybe I should check that.